.github/workflows/test.yml

```diff
@@ -11,78 +11,38 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
+      - run: rustup toolchain install stable --profile minimal
       - uses: Swatinem/rust-cache@v2
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --all-targets
+      - run: cargo check --all-targets
 
   test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
+      - run: rustup toolchain install stable --profile minimal
       - uses: Swatinem/rust-cache@v2
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: -- --test-threads=1 --ignored
-
+      - run: cargo test -- --test-threads=1 --ignored
 
   clippy:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-          components: clippy
+      - run: rustup toolchain install stable --profile minimal
       - uses: Swatinem/rust-cache@v2
-      - uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --all-targets -- -D warnings
-
+      - run: cargo clippy --all-targets -- -D warnings
 
   rustfmt:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly
-          override: true
-          components: rustfmt
+      - run: rustup toolchain install stable --profile minimal
       - uses: Swatinem/rust-cache@v2
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
-
+      - run: cargo fmt --all -- --check
```

Comment:
...can I ask if this runs with the nightly option? I don't know how to read `.yml` files, but I see line 65 (old file) specified nightly toolchain

Reply:
good catch! nope, this'll run on the `stable` toolchain with the changes.

Reply:
It's the same in a few other sections of the workflow (speaking of, do we still need the `nightly` test run?).

Reply:
Yeah, so I replaced some of the toolchains with the stable variant where no nightly options were used (from what I could tell at least).

Reply:
Was there a particular reason for using nightly on `cargo fmt`?

Reply:
We have fmt config that uses nightly options: https://github.com/runziggurat/zcash/blob/main/.rustfmt.toml
That being said, I don't know if they're still only available on nightly, perhaps they've been stabilised in the interim, worth checking.

Reply:
Does not look to be part of stable just yet. I'll push up a fix, it's only `cargo fmt` that needs nightly, right?

Reply:
should be yes, stable `clippy` is fine 👍 
We had some `nightly` workflows (tests and all) there too, not sure if they are still needed.

Reply:
I think the only actual necessary ones are for `cargo fmt` & for accessing the `-Z` flags during testing, so we should be all good. Ready for merge you think?